draft-irtf-cfrg-hpke.md

```diff
@@ -1155,26 +1253,93 @@ necessary to prove their security. The results from {{CS01}} provide
+Auth mode satisfies the desired message confidentiality and sender
+authentication properties listed at the beginning of this section;
+it does not consider multiple messages, nor the secret export API.
+
+### Post-Quantum Security
+
+All of {{CS01}}, {{HPKEAnalysis}}, and {{ABHKLR20}} are premised on
 classical security models and assumptions, and do not consider
 adversaries capable of quantum computation. A full proof of post-quantum
 security would need to take appropriate security models and assumptions
-into account, in addition to simply using a post-quantum KEM. The hybrid
-quantum-resistance property described above, which is achieved by using
-the PSK or AuthPSK mode, is proven in {{HPKEAnalysis}}; in a quantum
-setting, the remaining security level is smaller and defined by the
-post-quantum security level of the AEAD scheme.
-
-## Security Requirements on a KEM used within HPKE
-
-A KEM used within HPKE MUST ensure the following to avoid identity
-mis-binding issues: The KEM shared secret computed by `Encap()` and `Decap()` MUST
-depend explicitly on the KEM public key `pkR` and the encapsulated key `enc`,
-as observed in {{S01}}. The KEM shared secret returned by `AuthEncap()` and `AuthDecap()`
-MUST explicitly depend on the KEM public keys `pkR` and `pkS` and the encapsulated
-key `enc`. This is usually implemented by including these values explicitly into
-the context of the key derivation function used to compute the KEM shared
-secret. This is also how DHKEM meets the requirement.
+into account, in addition to simply using a post-quantum KEM. However,
+the composition theorems from {{ABHKLR20}} for HPKE's Auth mode only make
+standard assumptions (i.e., no random oracle assumption) that are expected
+to hold against quantum adversaries (although with slightly worse bounds).
+Thus, these composition theorems, in combination with a post-quantum-secure
+authenticated KEM, guarantee the post-quantum security of HPKE's Auth mode
+(as a reminder, DHKEM is not post-quantum-secure).
+In future work, the analysis from {{ABHKLR20}} can be extended to cover
+HPKE's other modes and desired security properties.
+The hybrid quantum-resistance property described above, which is achieved
+by using the PSK or AuthPSK mode, is not proven in {{HPKEAnalysis}} because
+this analysis requires the random oracle model; in a quantum
+setting, this model needs adaption to, for example, the quantum random
+oracle model.
+
+## Security Requirements on a KEM used within HPKE {#kem-security}
+
+A KEM used within HPKE MUST allow HPKE to satisfy its desired security
+properties described in {{sec-properties}}. In particular, the KEM
+shared secret MUST be a uniformly random byte string of length `Nsecret`.
+(This means, for instance, that it would not be sufficient if the KEM
+shared secret is only uniformly random as an element of some set prior
+to its encoding as byte string.)
```

Comment:
```suggestion
This means, for instance, that it would not be sufficient if the KEM
shared secret is only uniformly random as an element of some set prior
to its encoding as byte string.
```